grails-app/controllers/org/pih/warehouse/api/CombinedShipmentItemApiController.groovy

```diff
@@ -24,6 +24,23 @@ class CombinedShipmentItemApiController {
 
     def orderService
 
+    def getProductsInOrders = {
+        def minLength = grailsApplication.config.openboxes.typeahead.minLength
+        def vendor = Location.get(params.vendor)
+        def destination = Location.get(params.destination)
+        String[] terms = params?.name?.split(",| ")?.findAll { it }
+        if (params.name && params.name.size() < minLength) {
+            render([data: []])
+            return
+        }
+        def orderItems = orderService.getProductsInOrders(terms, destination, vendor)
+        def products = []
+        orderItems*.findAll{ it.orderItemStatusCode != OrderItemStatusCode.CANCELED && it.getQuantityRemainingToShip() > 0 }.each {
```

Comment:
Why aren't we handling this in the getProductsInOrders() query? Also we should be handling the "unique products" selection in the query as well.

Reply:
@jmiranda we can't handle this part "it.getQuantityRemainingToShip() > 0 " in createCriteria because this field is not in database, it's transient field.